### ISSUE TITLE
Where can I find the pytorch code that cause exceptions?

### ISSUE THREAD

**Issue by JohnHerry** [NVIDIA/TensorRT#4522]
I  am tring to convert an  AMP trained pytorch checkpoint to onnx and then tensorrt fp16,  all process are ok, but when loading the tensorrt fp16 model for inference, it will generate `Nan` values.   
I had use `polygraphy` to make test    
```
 polygraphy run --model=onnx  mymodel.fp32.onnx.simplified  \    
    --trt --fp16  \    
    --trt-min-shapes x:[2,80,4] mask:[2,1,4] cond:[2.80,4]   \   
    --trt-opt-shapes x:[2,80,192] mask:[2,1,192] cond:[2,80,192]  \   
    --trt-max-shapes x:[2,80, 1000] mask:[2,1,1000] cond:[2,80,1000]  \   
    --save_outputs layer_outputs.json --trt-outputs mark all
```    
the polygraphy report many errors.    
```
[E] 9: Skipping tactic 0x0000000000 due to exception [shape.cpp:verify_output_type:1308] Mismatched type for tensor ONNXTRT_Broadcast_39_output', fp16 vs. expected type:fp32   
[E]10: Could not find any implementation for node {ForeignNode[/dit_blocks.15/attn/Cast_8...ONNXTRT_unsqueezeTensor_12228]}
[E] 10 [optimizer.cpp::computeCosts::4105] Error Code 10: Internal Error (Could not find any implementation for node {ForeignNode[/dit_blocks.15/attn/Cast_8...ONNXTRT_unsqueezeTensor_12228]}.)
[!] Invalid Engine. Please ensure the engine was built correctly
```  

my onnxruntime opset_version=18,  and the exportion is : 
```
torch.onnx.export(
       estimator, 
       (x, mask, cond),
       "my.onnx",
       export_params=True,
       opset_version=18,
       do_constant_folding=True,
       input_names = ['x', 'mask', 'cond', 't', 'spks'],
       output_names=['estimator_output'],
       dynamic_axes={'x': {2: 'seq_len'},  'mask':{2: 'seq_len'}, 'cond': {2: 'seq_len'}, 'estimator_output': {2: 'seq_len'}}
)
```   
My question is ,  the polygraphy output tell that there are something wrong with my onnx exported model. but How do I convert the error message there to the right python code of model define or training pipeline?

Tensorrt V9.2   V10.0.1.6  are tested.

**Comment by JohnHerry** [comment]
It is strange that I have 16 dit_blocks, from dit_blocks.0  to dit_blocks.15;  the exception happed on the last one. and when I use polygrapy to run the default float32, the result is ok.  only happend on the  --fp16  opened.  while my model is trained using AMP    
and from the view of netron picture, the max unsqueeze node in my onnx model is onnx::Unsqueeze_7670;   no such node of ONNXTRTUnsqueezeTensor_12228

**Comment by JohnHerry** [comment]
It seems that dit_blocks.15  Cast_8 is not a root node of subtree,  the subtree root node is Cast_4   
the picture below, Cast_8 is the one in the red circle.

![Image](https://github.com/user-attachments/assets/ace9255e-67fb-4e58-ab74-d878273e8a89)

**Comment by asfiyab-nvidia** [comment]
@JohnHerry Can you please share the ONNX model or the export script so I can reproduce this issue on my end?

**Comment by JohnHerry** [comment]
the  model export script is just like[ here.](https://github.com/FunAudioLLM/CosyVoice/blob/v2.0/cosyvoice/bin/export_onnx.py),  and the tensorrt exporter is just like [here](https://github.com/FunAudioLLM/CosyVoice/blob/dev/Comet/cosyvoice/bin/export_trt.sh).  we had used the fp16 exporter.  as to the onnx, I am sorry for the company policy forbidden, but it is something like the DiT based flow-matching model mensioned[ in the paper](https://arxiv.org/abs/2505.17589), and there are 16 DiT blocks ，  The DiT block source code， inside which the exception happened, is this [DiTBlock](https://github.com/SWivid/F5-TTS/blob/main/src/f5_tts/model/modules.py#L648).

**Comment by asfiyab-nvidia** [comment]
@JohnHerry Setting up the environment to export this model is not trivial so here's my response just based on the information you've provided. 
I see you're using TRT 9.2. Can you please try again with 10.13 - The Polygraphy error may be resolved with the latest version which will have additional op support

**Comment by JohnHerry** [comment]
Thanks a lot, we will try.   and there are some additional info.
We had tried to export the onnx into tensorrt fp16 engine, but this engine will always generate Nan inference results.   
while we had tried the onnx model on inference ,there is no problem.  So we guess there should be some node in the trt model causing  numeric overflow.   We had tried to set the precision of layers[and there outputs] with types like `trt.LayerType.SOFTMAX,  trt.LayerType.REDUCE, trt.LayerType.ELEMENTWISE, trt.LayerType.ACTIVATION, trt.LayerType.UNARY` that may cause value overflow, as trt.DataType.FLOAT to prevent tensorrt optimizer change type into fp16, but with no luck, the engine will still generate Nan.    We also tried the `config.set_flag(trt.BuilderFlag.OBEY_PRECISION_CONSTRAINS)` but with no luck too.
